src/vs/workbench/contrib/files/browser/files.contribution.ts

```diff
@@ -323,6 +323,11 @@ configurationRegistry.registerConfiguration({
 			'default': 'askUser',
 			'scope': ConfigurationScope.LANGUAGE_OVERRIDABLE
 		},
+		'files.dialog.homePath': {
+			'type': 'string',
+			'description': nls.localize('fileDialogHomePath', "Default path for file dialogs, overriding user's home path."),
```

Comment:
It might be worth calling out in the description that default is first based on the last folder or file you had open and only then if that doesn't exist will this setting be uses.

Reply:
Good idea. I have pushed a change.

Reply:
I have also pushed a change so `~` replacement uses the true home.